### PR TITLE
Issue #641 - Use PackedSequence in RNNcore to improve pooling accuracy (and RNN speed)

### PR DESCRIPTION
**Pull request by ranihorev**
In the current implementation, the RNN output pooling calculates the average of the entire tensor including the padding. This pooling creates inconsistent behavior when predicting batches of texts (due to the change in the amount of padding). 
This PR replaces the previous padding with PyTorch PackedSequence to pack and pad the texts in the RNN core. This change also improves the RNN speed.
We also use the lengths to calculate the average pooling while ignoring the padding; 

Note: In the current implementation of MultiBatchRNN, predicting a batch that contains texts with more than 1*bptt difference in length is not recommended.

p.s.
It's my first PR, so feedback will be appreciated :)

### CODE REVIEW

**Comment by sgugger** [comment]
You made changes to RNNCore, so I expect it's going to break the language model. Did you test you could run the text example notebook entirely? In any case I'll need to check it doesn't change the final result of ULMFiT on IMDB.

For the future, please note that we prefer to get significant changes in the form of a new notebook like [the development notebooks](https://github.com/fastai/fastai_docs/tree/master/dev_nb) used to create the library, so we can easily refactor/check it works properly.

**Comment by jph00** [comment]
Thanks so much for this really interesting contribution! :) As @sgugger mentioned, it is breaking existing code, so we're not going to accept it in this state - and we'd like to be able to play around with your code in a notebook environment so we can fully understand it.

Therefore, would you be able to create a notebook which contains your implementation, along with examples showing how it works with language model and classification training and fine-tuning? Please create a gist and post it at forums.fast.ai so we can discuss.

Many thanks!

**Comment by ranihorev** [comment]
Sorry for the late response. 
It does break the language model (I updated the classifier but forgot about the LM...). Would you prefer a different input format that supports both versions? e.g. set the text lengths input as optional?
Anyway, I'll post my notebook sometime this week.
